docs/DesignMDIngestion.md

```diff
@@ -0,0 +1,116 @@
+# Generic Metadata Ingestion to MongoDB
+
+## Description
+
+This is the design document discussing the ingestion process of existing metadata
+into Zenko with MongoDB.
+
+The primary use case is to copy the existing metadata from the RING and S3 Connector
+and ingest into the MongoDB backend that is used with Zenko; future use cases will
+include copying metadata from existing storage solutions on AWS or Azure to MongoDB.
+
+This specific development will allow Zenko instances to copy and ingest existing
+metadata,
+copying the information from the raft logs completely and allowing MongoDB to be
+a parallel metadata database.
+
+## Purpose
+
+* Copy all existing metadata from pre-existing storage solutions to MongoDB in Zenko.
+
+* Allow MongoDB to be used in parallel with the existing metadata servers with S3
+  Connector, so that the primary metadata backend can be changed between the metadata
+  servers and the MongoDB backend.
+
+## Design
+
+The proposed design will be as follows:
+
+* First, determine if this is a fresh install of Zenko or an upgrade installation.
+    * This can be determined by checking if Zookeeper has stored a sequence ID.
+* If this is a fresh install of Zenko, we will create a `snapshot`.
```

Comment:
I assume it's a snapshot of the existing metadata from the "RING and S3 connector". Am I right?

Reply:
I think it is just getting the current log offset.

Reply:
The snapshot is getting log offset as well as listing all existing objects in current buckets - I will make this more clear in the documentation